examples/chatgpt/gpt_actions_library/gpt_action_mongodb_vector.md

```diff
@@ -0,0 +1,356 @@
+# GPT Action Library: MongoDB Atlas Vector Search
+
+## Introduction
+
+
+This page provides an instruction & guide for developers building a GPT Action for a specific application. Before you proceed, make sure to first familiarize yourself with the following information:
+
+- [Introduction to GPT Actions](https://platform.openai.com/docs/actions)
+- [Introduction to GPT Actions Library](https://platform.openai.com/docs/actions/actions-library)
+- [Example of Building a GPT Action from Scratch](https://platform.openai.com/docs/actions/getting-started)
+
+This particular GPT Action provides an overview of how to connect to MongoDB Atlas. This Action takes a user’s question and generates MQL to perform a semantic search on a particlar collection to answer the user’s question. It uses Google Cloud Run Functions and OpenAI's Embeddings API to transform the user query to an embedding that will be used as a query parameter for the semantic search.
+
+### Value + Example Business Use Cases
+
+**Value**: Users can now leverage ChatGPT's natural language capability to connect directly to MongoDB Atlas and perform a semantic search. 
+
+**Example Use Cases**:
+
+- A user needs to perform a vector search in MongoDB, but needs a middleware app between ChatGPT and Google Cloud SQL
+
+- A user has thousands of files and has gone through the process of vectorizing the files and storing them in MongoDB
+
+## Application Information
+
+### Application Key Links
+
+Check out these links before you get started:
+
+- OpenAI Embeddings API - https://platform.openai.com/docs/guides/embeddings
+- MongoDB Atlas Vector Search - https://www.mongodb.com/docs/atlas/atlas-vector-search/vector-search-overview/
+
+### Application Prerequisites
+
+Before you get started, make sure you go through the following steps in your application environment:
+
+- An OpenAI API key
+- Access to a MongoDB Atlas cluster
+- Google Cloud Console with access to create Google Cloud Run Functions and Google Cloud APIs (you will need this to set up the OAuth Client)
+
+## Middleware Information
+
+### Middleware Function
+
+To create a Google Run Function, follow the steps in the [GCP Functions Action](https://cookbook.openai.com/examples/chatgpt/gpt_actions_library/gpt_middleware_google_cloud_function) cookbook.
```

Comment:
would rename to "GPT Actions library (Middleware) - Google Cloud Function" (or other providers – we don't require GCP, right?)